solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -1682,7 +1684,6 @@ private void getDocListC(QueryResult qr, QueryCommand cmd) throws IOException {
         }
       }
     } else {
-      fullSortCount.increment();
```

Comment:
nocommit -- to move elsewhere still

Reply:
I looked at moving this to a better place inside `getDocListAndSetNC` as well as `getDocListNC` but it gets really complicated figuring out if a fullsortcount should be incremented or not.

Reply:
In those methods, I think it goes first on the `else` side of the condition with comment `// handle zero case...` thus when we need to return a doc list.